reader_concurrency_semaphore.cc

```diff
@@ -747,19 +747,19 @@ struct stop_execution_loop {
 future<> reader_concurrency_semaphore::execution_loop() noexcept {
     while (!_stopped) {
         try {
-            co_await _ready_list.not_empty();
+            co_await _ready_list_cv.wait();
```

Comment:
Please use condition_variable::when() (an allocation saved is an allocation earned)

Reply:
And check the series with `scylla perf_simple_query`. I think it will show a small improvement.

Reply:
Done in v2.